src/pages/settings/server/Roles.tsx

```diff
@@ -1,289 +1,223 @@
-                )}
-            </div>
-        </div>
+                    }
+
+                    if (!isEqual(current, value)) {
+                        server.editRole(selected, value);
+                    }
+                }
+
+                // Delete the role from this server.
+                function deleteRole() {
+                    server.deleteRole(selected);
+                }
+
+                return (
+                    <div>
+                        <SpaceBetween>
+                            <H1>Edit {currentRole.name}</H1>
+                            <Button
+                                palette="secondary"
+                                disabled={isEqual(
+                                    currentRole,
+                                    currentRoleValue,
+                                )}
+                                onClick={save}>
+                                Save
+                            </Button>
+                        </SpaceBetween>
+                        <hr />
+                        {selected !== "default" && (
+                            <>
+                                <section>
+                                    <Overline type="subtle">Role Name</Overline>
+                                    <p>
+                                        <InputBox
+                                            value={currentRoleValue.name}
+                                            onChange={(e) =>
+                                                setValue({
+                                                    ...value,
+                                                    name: e.currentTarget.value,
+                                                })
+                                            }
+                                            contrast
+                                        />
+                                    </p>
+                                </section>
+                                <section>
+                                    <Overline type="subtle">
+                                        Role Colour
+                                    </Overline>
+                                    <p>
+                                        <ColourSwatches
+                                            value={
+                                                currentRoleValue.colour ??
+                                                "gray"
+                                            }
+                                            onChange={(colour) =>
+                                                setValue({ ...value, colour })
+                                            }
+                                        />
+                                    </p>
+                                </section>
+                                <section>
+                                    <Overline type="subtle">
+                                        Role Options
```

Comment:
And here (please tell me if there's a way to just have a comment apply to multiple specific lines)